std/traits.d

```diff
@@ -8087,3 +8089,97 @@ enum isCopyable(S) = is(typeof(
     static assert(isCopyable!int);
     static assert(isCopyable!(int[]));
 }
+
+/**
+ * Determines the type of the argument.
+ *
+ * Returns:
+ *   T[0] if it's a type, typeof(T[0]) if it's a value, void if it's a template.
+ *
+ */
+template TypeOf(T...)
+    if (T.length == 1 && (is(T[0]) || is(typeof(T[0]))))
+{
+    static if (is(T[0]))
+        alias TypeOf = T[0];
+    else static if (is(typeof(T[0])))
+        alias TypeOf = typeof(T[0]);
+}
+
+///
+@safe unittest
+{
+    assert(is(TypeOf!3 == int));
+    assert(is(TypeOf!int == int));
+    assert(is(TypeOf!([]) == void[]));
+    assert(!is(TypeOf!(mixin(__MODULE__))));
+    assert(is(TypeOf!TypeOf));
+}
+
+/**
+ * Get a list of the overloads for the given callable.
+ */
+template getOverloads(func...)
+    if (func.length == 1 && isCallable!func)
+{
+    alias p = AliasSeq!(__traits(parent, func))[0];
+    static if (is(p) || p.stringof[0 .. 7] == "module ")
+        alias getOverloads = AliasSeq!(__traits(getOverloads, __traits(parent, func), __traits(identifier, func)));
+    else
+        alias getOverloads = func;
+}
+
+version(unittest)
+{
```

Comment:
These should instead be defined as static functions in the unit test block. Currently, the reader of the doc has no idea what's being tested.